### PR TITLE
Writing the Tables SDK for Java 

### PR DESCRIPTION
**Pull request by eboyd23**
Goal to write the Tables SDK for Java

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceAsyncClient.java [hunk 1]
```diff
@@ -504,7 +504,7 @@ public String getAccountUrl() {
      * @param properties Configures the service.
      * @return A {@link Mono} containing the storage account properties.
      */
-    public Mono<Void> setProperties(BlobServiceProperties properties) {
+    public Mono<Void> setProperties(BlobServiceProperties properties){
```
Comment (amishra-dev) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceAsyncClient.java:507]: please revert this change

File: sdk/tables/azure-data-tables/pom.xml [hunk 2]
```diff
@@ -42,6 +42,82 @@ Licensed under the MIT License.
...
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-core</artifactId>
+      <version>3.0.0</version> <!-- {x-version-update;org.mockito:mockito-core;external_dependency} -->
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-storage-common</artifactId>
+      <version>12.7.0</version>  <!-- {x-version-update;com.azure:azure-storage-common;dependency} -->
+    </dependency>
```
Comment (alzimmermsft) [inline sdk/tables/azure-data-tables/pom.xml:84]: This is listed twice

File: sdk/tables/azure-data-tables/pom.xml [hunk 3]
```diff
@@ -42,6 +42,82 @@ Licensed under the MIT License.
...
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-core-test</artifactId>
+      <version>1.3.0</version> <!-- {x-version-update;com.azure:azure-core-test;dependency} -->
```
Comment (alzimmermsft) [inline sdk/tables/azure-data-tables/pom.xml:88]: Version should be `1.3.1`. Unless the PR isn't up to date with upstream.

File: sdk/tables/azure-data-tables/pom.xml [hunk 4]
```diff
@@ -42,6 +42,82 @@ Licensed under the MIT License.
...
+      <artifactId>reactor-test</artifactId>
+      <version>3.3.5.RELEASE</version> <!-- {x-version-update;io.projectreactor:reactor-test;external_dependency} -->
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-core</artifactId>
+      <version>3.0.0</version> <!-- {x-version-update;org.mockito:mockito-core;external_dependency} -->
+      <scope>test</scope>
+    </dependency>
```
Comment (alzimmermsft) [inline sdk/tables/azure-data-tables/pom.xml:120]: Duplicate listings.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/AzureTable.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables;
+
+/**
+ * class for a table object
+ */
+public class AzureTable {
+    private final String name;
+    private TableClient tableClient;
```
Comment (alzimmermsft) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/AzureTable.java:10]: This will be set as some point, correct?

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/ContinuationToken.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package com.azure.data.tables;
```
Comment (alzimmermsft) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/ContinuationToken.java:1]: Missing license and documentation.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/ContinuationToken.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+public class ContinuationToken {
+    String nextPartitionKey;
```
Comment (alzimmermsft) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/ContinuationToken.java:4]: Should this and `nextRowKey` be `private final`

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java [hunk 8]
```diff
@@ -0,0 +1,484 @@
...
+    TableAsyncClient(String tableName) {
+        this.tableName = tableName;
+    }
+
+    /**
+     * returns the table name associated with the client
+     *
+     * @return table name
+     */
+    public Mono<String> getTableName() {
```
Comment (alzimmermsft) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java:49]: Should this be a synchronous method as it is returning a property about the client?

```java
public String getTableName() {
  return tableName;
}
```

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java [hunk 9]
```diff
@@ -0,0 +1,484 @@
...
+                    return new TableEntity(e);
+                })
+                .collect(Collectors.toList());
+
+            try {
+                return Mono.just();
+                FeedPage
+            }
+
+        })
```
Comment (alzimmermsft) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java:89]: Missing a `;`

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java [hunk 10]
```diff
@@ -0,0 +1,484 @@
...
+        return new PagedFlux<TableEntity>(
+            () -> withContext(context -> queryFistPageEntities(queryOptions, context)),
+            continuationToken -> withContext(context -> queryNextPageEntities(context, continuationToken.getNextPartitionKey(), nextRowKey))
+        );
+
+    }
+
+    private Mono<PagedResponse<TableEntity>> queryFirstPageEntities(QueryOptions queryOptions, Context context) {
+        String requestID = "";
+        return impl.queryEntitiesWithResponseAsync(tableName, TIMEOUT, requestID, null, null,
```
Comment (alzimmermsft) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java:71]: Should be safe passing `null` instead of `""` for request ID.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java [hunk 11]
```diff
@@ -0,0 +1,484 @@
...
+ * class for the table async client
+ */
+@ServiceClient(
+    builder = TableClientBuilder.class,
+    isAsync = true)
+public class TableAsyncClient {
+    private final ClientLogger logger = new ClientLogger(TableAsyncClient.class);
+    private final String tableName;
+    private final TablesImpl impl = null;
+    private final Integer TIMEOUT = 400;
```
Comment (alzimmermsft) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java:38]: Should this be a configuration value passed by a customer? I'm guessing if nothing is passed the operation has no timeout. Also is this in seconds, milliseconds, etc?

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceAsyncClient.java (hunk 1 above)
Comment (eboyd23) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceAsyncClient.java]: So sorry, must have miss-clicked! Fixed now.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java [hunk 12]
```diff
@@ -2,105 +2,566 @@
...
+        } else {
+            if (ifUnchanged) {
+                return tableImplementation.updateEntityWithResponseAsync(tableName, entity.getPartitionKey().toString(),
+                    entity.getRowKey().toString(), timeoutInt, null, entity.getETag(), entity.getProperties(),
+                    null, context).map(response -> {
+                        return new SimpleResponse<>(response.getRequest(), response.getStatusCode(),
+                            response.getHeaders(), null);
+                    });
+            } else {
+                return getEntity(entity.getPartitionKey().toString(), entity.getRowKey().toString())
```
Comment (eboyd23) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java:323]: @conniey here is the part which I was not sure how to do without a get.

File: sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/TablesAsyncClientTest.java [hunk 13]
```diff
@@ -0,0 +1,337 @@
...
+            })
+            .expectComplete()
+            .verify();
+    }
+
+    /**
+     * expect to see both propertyA and propertyB since the UpdateMode is MERGE
+     */
+    @Test
+    void updateEntityWithResponseMergeUnconditional() {
```
Comment (eboyd23) [inline sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/TablesAsyncClientTest.java:214]: @conniey This test is failing because the get isn't working, I could delete these tests until we get it fixed?

File: sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/TableServiceAsyncClientTest.java [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+            })
+            .expectComplete()
+            .verify();
+
+    }
+
+    @Test
+    void listTableWithResponseWithParams() {
+        // Arrange
+        QueryParams queryParams = new QueryParams().setFilter("TableName eq SampleTable");
```
Comment (eboyd23) [inline sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/TableServiceAsyncClientTest.java:135]: having trouble getting this filter to work

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/Entity.java [hunk 15]
```diff
@@ -0,0 +1,114 @@
...
+    /**
+     * gets the etag
+     *
+     * @return the etag for the entity
+     */
+    public String getETag() {
+        return etag;
+    }
+
+    public Entity setEtag(String etag) {
```
Comment (eboyd23) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/Entity.java:88]: @conniey Need to get this to be private but accessible to the service client

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/BuilderHelper.java [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+import com.azure.storage.common.policy.RequestRetryPolicy;
+import com.azure.storage.common.policy.ResponseValidationPolicyBuilder;
+import com.azure.storage.common.policy.ScrubEtagPolicy;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+class BuilderHelper {
+    private static final Map<String, String> PROPERTIES =
+        CoreUtils.getProperties("azure-storage-blob.properties");
```
Comment (conniey) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/BuilderHelper.java:34]: This should be azure-data-tables.properties
